contracts/mixnet/src/families/transactions.rs

```diff
@@ -0,0 +1,391 @@
+            owner: owner.clone(),
+        })?
+        .1;
+
+    ensure_bonded(&existing_bond)?;
+
+    validate_family_signature(
+        deps.as_ref(),
+        existing_bond.identity(),
+        &signature,
+        family_head.identity(),
+    )?;
+
+    let mut family = Family::get(&family_head, proxy, deps.storage)?;
+
+    family.add(deps.storage, existing_bond.identity())?;
+
+    Ok(Response::default())
+}
+
+pub fn try_leave_family(
+    deps: DepsMut,
+    info: MessageInfo,
+    signature: String,
+    family_head: IdentityKey,
+) -> Result<Response, MixnetContractError> {
+    let family_head = FamilyHead::new(&family_head);
+    _try_leave_family(deps, &info.sender, signature, family_head, None)
+}
+
+pub fn try_leave_family_on_behalf(
+    deps: DepsMut,
+    info: MessageInfo,
+    member_address: String,
+    signature: String,
+    family_head: IdentityKey,
+) -> Result<Response, MixnetContractError> {
+    let family_head = FamilyHead::new(&family_head);
+    let member_address = deps.api.addr_validate(&member_address)?;
+    _try_leave_family(
+        deps,
+        &member_address,
+        signature,
+        family_head,
+        Some(info.sender),
+    )
+}
+
+fn _try_leave_family(
+    deps: DepsMut,
+    owner: &Addr,
+    signature: String,
+    family_head: FamilyHead,
+    proxy: Option<Addr>,
+) -> Result<Response, MixnetContractError> {
+    let existing_bond = crate::mixnodes::storage::mixnode_bonds()
+        .idx
+        .owner
+        .item(deps.storage, owner.clone())?
+        .ok_or(MixnetContractError::NoAssociatedMixNodeBond {
+            owner: owner.clone(),
+        })?
+        .1;
+
+    ensure_bonded(&existing_bond)?;
+
+    validate_family_signature(
+        deps.as_ref(),
+        existing_bond.identity(),
+        &signature,
+        family_head.identity(),
+    )?;
+
+    let mut family = Family::get(&family_head, proxy, deps.storage)?;
+
+    family.remove(deps.storage, existing_bond.identity())?;
+
+    Ok(Response::default())
+}
+
+pub fn try_head_kick_member(
+    deps: DepsMut,
+    info: MessageInfo,
+    owner_signature: String,
+    member: IdentityKeyRef,
+) -> Result<Response, MixnetContractError> {
+    _try_head_kick_member(deps, &info.sender, owner_signature, member, None)
+}
+
+pub fn try_head_kick_member_on_behalf(
+    deps: DepsMut,
+    info: MessageInfo,
+    head_address: String,
+    owner_signature: String,
+    member: IdentityKeyRef,
+) -> Result<Response, MixnetContractError> {
+    let head_address = deps.api.addr_validate(&head_address)?;
+    _try_head_kick_member(
+        deps,
+        &head_address,
+        owner_signature,
+        member,
+        Some(info.sender),
+    )
+}
+
+fn _try_head_kick_member(
+    deps: DepsMut,
+    owner: &Addr,
+    owner_signature: String,
+    member: IdentityKeyRef<'_>,
+    proxy: Option<Addr>,
+) -> Result<Response, MixnetContractError> {
+    let existing_bond = crate::mixnodes::storage::mixnode_bonds()
```

Comment:
thinking about it, don't we need to check whether the existing bond actually has the same identity as the family head? (i.e. what would have happened if you created family under identity `X`, unbonded and then rebonded with identity `Y`?)

Reply:
Great call, we should totally check

Reply:
Altough, looking at the implementation, if the identity changed, the head is no longer head, so it can't kick anyone out